### PR TITLE
Update dependencies, Update files upload, add background resources router

### PR DESCRIPTION
**Pull request by Tesseract22**
Use `@volar-plugins/vetur` instead of `Vetur` as suggested. Update `packages.json` and `vite.config.ts` to load the new package.

Also make minor changes to files toolbar.

### CODE REVIEW

File: src/components/NavbarTemp.vue [hunk 1]
```diff
@@ -84,50 +84,46 @@ li b-button:hover {
...
-  display: block;
-  color: rgb(0, 0, 0);
-  margin: 0;
-  text-align: center;
-  padding-top: 18px;
-  padding-right: 20px;
-  padding-bottom: 20px;
-  padding-left: 20px;
-  text-decoration: none;
+    display: block;
```
Comment (TwinIsland) [inline src/components/NavbarTemp.vue:101]: conflict with main branch need to be resolved

**Review by TwinIsland — CHANGES_REQUESTED** [review]
fix the conflict with main branch

File: src/components/NavbarTemp.vue (hunk 1 above)
Comment (Tesseract22) [reply src/components/NavbarTemp.vue]: Fix conflicts

**Review by TwinIsland — APPROVED** [review]

**Review by TwinIsland — APPROVED** [review]
all set, merge approved

**Pull request merged**